llvm/lib/Transforms/Scalar/LoopStrengthReduce.cpp

```diff
@@ -6811,7 +6811,7 @@ canFoldTermCondOfLoop(Loop *L, ScalarEvolution &SE, DominatorTree &DT,
   // If ToFold does not have an incoming value from LoopLatch then the simple
```

Comment:
Your comment about unreachability is a bit misleading - the reachability of the other loop isn't the key detail.  Instead, maybe something along the lines of: Make sure the recurrence matched is part of the current loop of interest.

Reply:
Updated comment and order of comparison to match the new comment.

Also updated PR message to be the squashed commit message.